cli/docker/environment_source.go

```diff
@@ -0,0 +1,56 @@
+package docker
+
+import (
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/log"
+)
+
+// implementing env.EnvironmentSource
+type DockerEnvironmentSource struct {
+	Logger log.Logger
+}
+
+func (des *DockerEnvironmentSource) GetEnvironment() map[string]string {
+	passthroughEnvsList := strings.Split(os.Getenv("BITRISE_DOCKER_PASSTHROUGH_ENVS"), ",")
+	passthroughEnvsList = append(passthroughEnvsList, "PATH", "PR", "ENVMAN_ENVSTORE_PATH")
+	dockerPassthroughEnvsMap := make(map[string]bool)
+	for _, k := range passthroughEnvsList {
+		dockerPassthroughEnvsMap[k] = true
+	}
+
+	processEnvs := os.Environ()
+	envs := make(map[string]string)
+
+	// String names can be duplicated (on Unix), and the Go libraries return the first instance of them:
+	// https://github.com/golang/go/blob/98d20fb23551a7ab900fcfe9d25fd9cb6a98a07f/src/syscall/env_unix.go#L45
+	// From https://pubs.opengroup.org/onlinepubs/9699919799/:
+	// > "There is no meaning associated with the order of strings in the environment.
+	// > If more than one string in an environment of a process has the same name, the consequences are undefined."
+	for _, env := range processEnvs {
+		key, value := des.splitEnv(env)
+		_, allowed := dockerPassthroughEnvsMap[key]
+		if !strings.HasPrefix(key, "BITRISE") && (key == "" || !allowed) {
```

Comment:
Did you review the full list of Bitrise-defined envs? I can't think of any exceptions now, but I want to make sure we don't miss any edge cases.
https://devcenter.bitrise.io/en/references/available-environment-variables.html

Reply:
The BITRISE prefix is sort of like a insurance to make sure those will go through as a minimum, we have looked at that list and didn't find anything problematic. The allowlist introduces via the passthrough variable is on top of this.